executor/index_merge_reader.go

```diff
@@ -605,8 +659,25 @@ func (w *partialIndexWorker) extractTaskHandles(ctx context.Context, chk *chunk.
 			return handles, retChk, nil
 		}
 		for i := 0; i < chk.NumRows(); i++ {
-			h := kv.IntHandle(chk.GetRow(i).GetInt64(handleOffset))
-			handles = append(handles, h)
+			var handle kv.Handle
+			if handleCols.IsInt() {
```

Comment:
We can use `handleCols.BuildHandle`.

Reply:
The input handleCols's schema may not be the index schema, but we can use `ResolveIndices` to build index schema handleCols.